packages/core/src/components/Button/Button.tsx

```diff
@@ -79,24 +95,30 @@ const mapVariant = (
 /**
  * Button component is used to trigger an action or event.
  */
-export const HvButton = forwardRef<HTMLButtonElement, HvButtonProps>(
-  (props, ref) => {
+export const HvButton: <C extends React.ElementType = "button">(
+  props: HvButtonProps<C>
+) => React.ReactElement | null = forwardRef(
+  <C extends React.ElementType = "button">(
+    props: HvButtonProps<C>,
+    ref: PolymorphicRef<C>
+  ) => {
     const {
       id,
       classes,
       children,
       variant = "primary",
       onClick,
-      disabled,
+      disabled = false,
       className,
       startIcon,
       endIcon,
       icon = false,
       size,
       radius = "base",
       overrideIconColors = true,
+      as: Component = "button",
```

Comment:
I do like `as`; it's shorter and clear enough, but we're already using the `component` prop in other components (same as MUI), including the previous `HvButton` (currently an undocumented breaking change)

I think we should keep the `component` prop name. What do you think?

```suggestion
      component: Component = "button",
```

Reply:
Okay, I will change our polymorphic types for use component `instead`

Reply:
The `as` property comes from the `PolymorphicComponent` type. This type is used by our `HvBox` component. If I change `as` to `component` this will lead to a breaking change for the `HvBox` component.

Reply:
But I don't believe the `HvBox` is currently being used to be honest but this will be a breaking change nonetheless.

Reply:
Maybe we can discuss this with the team and see what they think (if we should update `as` to `component`)? I will updated the migration guide afterwards.

Reply:
Sure 👍 

I wasn't aware of other components using it.

Well, `HvBox` doesn't have a documentation page, and it's barely used in out components and samples.

I think that we definitely should be leveraging `PolymorphicComponentRef` in other polymorphic components, so that we get the correct typings. So, to me this isn't a `HvBox` vs `HvButton`, but a `HvBox` vs many components 😅.

I think it makes more sense to just drop the `PolymorphicComponentRef` "feature" in `HvBox` and add it to the components (ie. Typography, Table, Avatar, etc.) that are already using a `component` prop and UI-Kit users commonly use. It wouldn't be a breaking change or change in API, just a drop in type safety. Alternatively, we could duplicate the type

Some examples of the issue we have:
```tsx
<HvTypography src="typography is not an img">
<HvAvatar component="a" href="asda" /> // does not accept `href`
```